File: src/main/java/com/hjj/apiserver/controller/AccountBookController.java
package com.hjj.apiserver.controller;

import com.hjj.apiserver.common.ApiError;
import com.hjj.apiserver.common.ApiResponse;
import com.hjj.apiserver.common.ApiUtils;
import com.hjj.apiserver.domain.UserEntity;
import com.hjj.apiserver.dto.AccountBookDto;
import com.hjj.apiserver.service.AccountBookService;
import com.hjj.apiserver.util.CurrentUser;
import io.swagger.annotations.*;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.modelmapper.ModelMapper;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;

@Api(tags = {"5. AccountBook"})
@Slf4j
@RestController
@AllArgsConstructor
public class AccountBookController {
    private final AccountBookService accountBookService;
    private final ModelMapper modelMapper;

    @ApiImplicitParams({
            @ApiImplicitParam(name = "Authorization", value = "로그인 성공 후 access_token", required = true, dataType = "String", paramType = "header")})
    @ApiOperation(value = "가계부 생성", notes = "가계부를 생성 한다.")
    @PostMapping("/account-book")
    public ApiResponse accountBookAdd(@CurrentUser UserEntity user, @Valid @RequestBody @ApiParam(value = "가계부 생성 객체", required = true) AccountBookDto.RequestAccountBookAddForm form) {
        try{
            AccountBookDto accountBookDto = modelMapper.map(form, AccountBookDto.class);

            accountBookService.addAccountBook(user, accountBookDto);

            return ApiUtils.success(null);
        }catch (Exception e){
            return ApiUtils.error(ApiError.ErrCode.ERR_CODE9999.getMsg(), ApiError.ErrCode.ERR_CODE9999);
        }
    }

    @ApiImplicitParams({
            @ApiImplicitParam(name = "Authorization", value = "로그인 성공 후 access_token", required = true, dataType = "String", paramType = "header")})
    @ApiOperation(value = "개인가계부 목록을 조회.", notes = "개인가계부를 조회 한다.")
    @GetMapping("/account-book")
    public ApiResponse accountBookList(@CurrentUser UserEntity user, @Valid AccountBookDto.RequestAccountBookFindAllForm form) {
        try{
            AccountBookDto accountBookDto = modelMapper.map(form, AccountBookDto.class);
            accountBookDto.setUserNo(user.getUserNo());
            return ApiUtils.success(accountBookService.findAllAccountBook(accountBookDto));
        }catch (Exception e){
            return ApiUtils.error(ApiError.ErrCode.ERR_CODE9999.getMsg(), ApiError.ErrCode.ERR_CODE9999);
        }
    }

    @ApiImplicitParams({
            @ApiImplicitParam(name = "Authorization", value = "로그인 성공 후 access_token", required = true, dataType = "String", paramType = "header")})
    @ApiOperation(value = "개인가계부 상세 조회.", notes = "개인가계부를 상세 조회 한다.")
    @GetMapping("/account-book/{accountBookNo}")
    public ApiResponse accountBookDetails(@CurrentUser UserEntity user, @PathVariable Long accountBookNo) {
        try{
            AccountBookDto accountBookDto = new AccountBookDto();
            accountBookDto.setUserNo(user.getUserNo());
            accountBookDto.setAccountBookNo(accountBookNo);

            return ApiUtils.success(accountBookService.findAccountBookDetail(accountBookDto));
        }catch (Exception e){
            return ApiUtils.error(ApiError.ErrCode.ERR_CODE9999.getMsg(), ApiError.ErrCode.ERR_CODE9999);
        }
    }
}

File: src/main/kotlin/com/hjj/apiserver/controller/AccountBookController.kt
package com.hjj.apiserver.controller

import com.hjj.apiserver.dto.accountbook.request.AccountBookAddRequest
import com.hjj.apiserver.dto.accountbook.response.AccountBookAddResponse
import com.hjj.apiserver.dto.accountbook.response.AccountBookDetailResponse
import com.hjj.apiserver.dto.accountbook.response.AccountBookFindAllResponse
import com.hjj.apiserver.dto.user.CurrentUserInfo
import com.hjj.apiserver.service.impl.AccountBookService
import com.hjj.apiserver.util.AuthUser
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
class AccountBookController(
    private val accountBookService: AccountBookService,
) {
    @PostMapping("/account-books")
    @ResponseStatus(HttpStatus.CREATED)
    fun accountBookAdd(
        @AuthUser authUserInfo: CurrentUserInfo,
        @Valid @RequestBody request: AccountBookAddRequest,
    ): AccountBookAddResponse {
        return accountBookService.addAccountBook(authUserInfo.userNo, request)
    }

    @GetMapping("/account-books")
    fun accountBooksFind(
        @AuthUser user: CurrentUserInfo,
    ): List<AccountBookFindAllResponse> {
        return accountBookService.findAllAccountBook(user.userNo)
    }

    @GetMapping("/account-books/{accountBookNo}")
    fun accountBookDetail(
        @AuthUser user: CurrentUserInfo,
        @PathVariable accountBookNo: Long,
    ): AccountBookDetailResponse {
        val findAccountBookDetail = accountBookService.findAccountBookDetail(accountBookNo, user.userNo)
        return findAccountBookDetail
    }
}
